packages/plugin-assets-retry/src/runtime/asyncChunkRetry.ts

```diff
@@ -164,30 +172,50 @@ function nextRetry(chunkId: string): Retry {
 const originalEnsureChunk = __RUNTIME_GLOBALS_ENSURE_CHUNK__;
 const originalGetChunkScriptFilename =
   __RUNTIME_GLOBALS_GET_CHUNK_SCRIPT_FILENAME__;
+const originalGetCssFilename =
+  __RUNTIME_GLOBALS_GET_MINI_CSS_EXTRACT_FILENAME__ ??
+  __RUNTIME_GLOBALS_GET_CSS_FILENAME__ ??
+  (() => null);
 const originalLoadScript = __RUNTIME_GLOBALS_LOAD_SCRIPT__;
 
 // if users want to support es5, add Promise polyfill first https://github.com/webpack/webpack/issues/12877
 function ensureChunk(chunkId: string): Promise<unknown> {
   const result = originalEnsureChunk(chunkId);
   const originalScriptFilename = originalGetChunkScriptFilename(chunkId);
+  const originalCssFilename = originalGetCssFilename(chunkId);
 
-  // mark the async chunk name in the global variables and share it with initial chunk retry
-  if (
-    typeof window !== 'undefined' &&
-    !window.__RB_ASYNC_CHUNKS__[originalScriptFilename]
-  ) {
-    window.__RB_ASYNC_CHUNKS__[originalScriptFilename] = true;
+  // mark the async chunk name in the global variables and share it with initial chunk retry to avoid duplicate retrying
+  if (typeof window !== 'undefined') {
+    if (
+      originalScriptFilename &&
+      !window.__RB_ASYNC_CHUNKS__[originalScriptFilename]
```

Comment:
Maybe we can remove `!window.__RB_ASYNC_CHUNKS__[originalScriptFilename]` to reduce runtime code?

Re-assigning `true` should not have side-effects.